src/commcare_cloud/fab/operations/staticfiles.py

```diff
@@ -97,18 +97,61 @@ def collectstatic(use_current_release=False):
 
 
 @parallel
-@roles(ROLES_STATIC)
+@roles(ROLES_STATIC_PRIMARY)
 def compress(use_current_release=False):
     """Run Django Compressor after a code update"""
     venv = env.py3_virtualenv_root if not use_current_release else env.py3_virtualenv_current
     with cd(env.code_root if not use_current_release else env.code_current):
         sudo('{}/bin/python manage.py compress --force -v 0'.format(venv))
         sudo('{}/bin/python manage.py purge_compressed_files'.format(venv))
-    update_manifest(save=True, use_current_release=use_current_release)
+
+    push_manifest(use_current_release=use_current_release)
+
+
+def push_manifest(use_current_release=False):
+    if env.use_shared_dir_for_staticfiles:
+        with cd(env.code_root if not use_current_release else env.code_current):
+            git_hash = _get_git_hash()
```

Comment:
what's the reasoning to use the git hash? why not use the release name? What happens if you deploy twice with the same git hash?

Reply:
Yeah I thought about that and couldn't figure out if one was clearly better than the other so I went with what the pre-existing process (the update_manifest management command) did and [used the git hash](https://github.com/dimagi/commcare-hq/blob/1b4a2a6c08ca3b50a4e6e86e2c57220a64f32b5c/corehq/apps/hqwebapp/management/commands/update_manifest.py#L61-L68).

I think if you deploy twice from the same hash the second time it'll overwrite the first time, but I think that's okay.